tests/admin_views/test_nav_sidebar.py

```diff
@@ -77,6 +81,25 @@ def test_sidebar_aria_current_page_missing_without_request_context_processor(sel
         self.assertContains(response, '<a href="%s">Users</a>' % url)
         self.assertNotContains(response, 'aria-current')
 
+    @override_settings(DEBUG=True)
+    def test_included_app_list_template_context_fully_set(self):
+        template_logger = logging.getLogger('django.template')
+        default_logger_level = template_logger.level
+        default_handler_level = self.logger.handlers[0].level
+
+        # Adjust levels on logger AND handler.
+        template_logger.level = logging.DEBUG
+        self.logger.handlers[0].level = logging.DEBUG
+
+        # All context variables should be set when rendering the sidebar.
+        url = reverse('test_with_sidebar:auth_user_changelist')
+        self.client.get(url)
+        self.assertEqual(self.logger_output.getvalue(), '')
+
+        # Reset levels
+        template_logger.level = default_logger_level
+        self.logger.handlers[0].level = default_handler_level
+
```

Comment:
You can simplify this with `assertLogs()`:
```suggestion
        url = reverse('test_with_sidebar:auth_user_changelist')
        with self.assertRaisesMessage(AssertionError, 'no logs'):
            with self.assertLogs('django.template', 'DEBUG'):
                self.client.get(url)
```

Reply:
Sneaky. I like it. 😀